pages/pil/dashboard/index.js

```diff
@@ -91,18 +91,22 @@ module.exports = settings => {
       res.locals.static.isAsru = req.user.profile.asruUser;
       res.locals.static.isLicensing = req.user.profile.asruLicensing;
 
-      // can only transfer own pil if it's active and has no in-progress amendment
-      const hasOpenAmendment = req.pil.status === 'active' && get(req.model, 'openTasks[0].data.action') === 'grant';
-      res.locals.static.canTransferPil = req.pil.status === 'active' && req.user.profile.id === req.profile.id && !hasOpenAmendment;
+      return req.user.can('pil.transfer', { pilId: req.pilId })
+        .then(canTransfer => {
+          const hasOpenAmendment = req.pil.status === 'active' && get(req.model, 'openTasks[0].data.action') === 'grant';
 
-      next();
+          res.locals.static.canTransferPil = req.pil.status === 'active' && !hasOpenAmendment && canTransfer;
+        })
```

Comment:
I'd be tempted to switch the order of these checks so that the call to `req.user.can` is only executed once the local checks have passed. This means we can avoid the extra API call in the case where we already know it's not transferable.

Reply:
I had it that way initially for exactly that reason but decided the tradeoff in better readability was worth the extra call. I can switch it back.

Reply:
It might just be the diff view, but it's not _super_ readable as it is, so I'd switch it back.

Might be worth pulling out into it's own function, especially since it looks a lot like there's a similar (but not identical) bit of logic in the `pil/establishment` router too 👇